### PR TITLE
[MSSQL] Add logs

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## Part Of https://github.com/elastic/connectors/issues/2299

Adding more logs in MSSQL connector.

Log File: https://drive.google.com/file/d/1Rnm-tMY-ECYv1JZpj3oP9SvMljZZRxjZ/view?usp=drive_link

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

**Comment by moxarth-elastic** [comment]
buildkite test this

File: connectors/sources/mssql.py [hunk 1]
```diff
@@ -255,6 +255,12 @@ async def ping(self):
     async def get_tables_to_fetch(self, is_filtering=False):
         tables = configured_tables(self.tables)
         if is_wildcard(tables) or is_filtering:
+            msg = (
+                "Fetching all tables as the configuration field 'tables' is set to '*'"
+                if not is_filtering
+                else "Fetching all tables as the advanced sync rules are enabled."
+            )
```
Comment (seanstory) [inline connectors/sources/mssql.py:262]:
```suggestion
            msg = "Fetching all tables"
```

File: connectors/sources/mssql.py [hunk 2]
```diff
@@ -496,6 +512,7 @@ def row2doc(self, row, doc_id, table, timestamp):
         return row
 
     async def get_primary_key(self, tables):
+        self._logger.debug(f"Extracting primary keys for tables: {tables}")
```
Comment (seanstory) [inline connectors/sources/mssql.py:515]: Lets log what we find as primary keys, too

File: connectors/sources/mssql.py [hunk 3]
```diff
@@ -302,9 +309,13 @@ async def get_table_primary_key(self, table):
                 retry_count=self.retry_count,
             )
         ]
+
+        self._logger.debug(f"Found primary keys for '{table}' table")
+
         return primary_keys
 
     async def get_table_last_update_time(self, table):
+        self._logger.debug(f"Fetching last updated time for table: {table}")
```
Comment (seanstory) [inline connectors/sources/mssql.py:318]: Let's log what time we found, too

**Review by seanstory — COMMENTED** [review]
Few nits, but mostly looks good.

**Comment by artem-shelkovnikov** [comment]
Copying from another PR - that's applicable to all SQL connectors:

The most useful connector logs are - which queries were actually executed by the connectors. Can you add debug logs that state which exact queries were executed? A good addition to it would be to also debug log the number of records returned by the query.

**Comment by moxarth-elastic** [comment]
buildkite test this

File: connectors/sources/mssql.py [hunk 4]
```diff
@@ -332,22 +346,31 @@ async def data_streamer(self, table=None, query=None):
...
+                cursor_query,
             ),
             fetch_columns=True,
             fetch_size=self.fetch_size,
             retry_count=self.retry_count,
         ):
+            record_count += 1
             yield data
 
+        self._logger.info(f"Found {record_count} records for '{cursor_query}' query")
```
[the remark this answers is not in the document]
Comment (artem-shelkovnikov) [reply connectors/sources/mssql.py:372]: How does this one look when multiline query is there? Does it render with `\n` or real line-breaks?

**Comment by moxarth-elastic** [comment]
buildkite test this

**Comment by artem-shelkovnikov** [comment]
@moxarth-elastic I'll be taking over this and some other PRs reviewed by me to finish the work

**Review by artem-shelkovnikov — DISMISSED** [review]

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**